### PR TITLE
Feature audio and video support

### PR DESCRIPTION
**Pull request by arubio**
Jira issue: https://vilua.atlassian.net/browse/SHAD-628

https://github.com/SignificoHealth/recco-ios-sdk/assets/2614596/0c09bdb7-041a-467e-8745-2e323b7cfca6

Added new cards UI to be able to tell the kind of content that a recommendation is related to.
Added audio and video recommendation detail. 
Added audio player to elegible articles with audio attached to them.
Added Picture in Picture support for eligible host apps. 
Video and Audio players will be synced with the system's now playing information when background audio/video is available.

### CODE REVIEW

File: Sources/ReccoHeadless/Entities/AppUserMedia.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+extension MediaType {
+    public init?(_ contentType: ContentType) {
+        switch contentType {
+        case .audio:
+            self = .audio
+        case .video:
+            self = .video
+        default:
+            return nil
```
[the remark this answers is not in the document]
Comment (arubio) [reply Sources/ReccoHeadless/Entities/AppUserMedia.swift:23]: this is more of an internal type, this only fails if we code the thing wrong, meaning, we try to cast an article content type to a media type. I use this to navigate, but it is not on the DTO layer or anything like that

File: Sources/ReccoUI/Media/MediaDetailView.swift [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        )
+        .reccoErrorView(
+            error: $viewModel.initialLoadError,
+            onRetry: {
+                await viewModel.initialLoad()
+            }
+        )
+        .background(Color.reccoBackground.ignoresSafeArea())
+        .reccoNotification(error: $viewModel.actionError)
+        .environment(\.currentScrollOffsetId, "\(self)")
```
[the remark this answers is not in the document]
Comment (arubio) [reply Sources/ReccoUI/Media/MediaDetailView.swift:96]: I need to check this, maybe it is not used anymore. It is a way of keeping up with scroll offsets in views that are contained in a scrollview, but the scrollview needs to emit that offset and its id (which in this case is a description of the view itself, with all its typed glory)

File: Sources/ReccoUI/Media/MediaDetailView.swift [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                        overlayView: {
+                            if viewModel.mediaType == .audio {
+                                image
+                            }
+                        }
+                    )
+                    .frame(height: headerHeight * 0.8)
+                    Spacer()
+                    Spacer()
+                    Spacer()
```
[the remark this answers is not in the document]
Comment (arubio) [reply Sources/ReccoUI/Media/MediaDetailView.swift:188]: that width needs to be dynamic depending on every screen height. I could do calculations or I could use spacer xD

File: Sources/ReccoUI/Resources/LocalResources/Assets.xcassets/Content/bookmark_filled.imageset/Contents.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "images" : [
     {
-      "filename" : "bookmark_filled.svg",
+      "filename" : "bookmark (2).svg",
```
[the remark this answers is not in the document]
Comment (arubio) [reply Sources/ReccoUI/Resources/LocalResources/Assets.xcassets/Content/bookmark_filled.imageset/Contents.json:4]: no, no, this is an internal xcode file, the names of the assets themselves have not changed, just the internal file name. Basically you never see this json anywhere, this is just xcode keeping track of internal stuff

**Review by chemarubio — APPROVED** [review]
🚀

**Pull request merged**